app/views/components/_markdown-editor.html.erb

```diff
@@ -1,17 +1,26 @@
+<%
+  textarea||= {}
+  edit_button_text||= 'Edit'
+  preview_button_text||= 'Preview'
+%>
 <div class="app-c-markdown-editor" data-module="markdown-editor">
+  <% if textarea[:label] %>
   <%= render "govuk_publishing_components/components/label", {
       text: textarea[:label][:text],
       html_for: textarea[:id]
   } %>
   <% textarea[:label] = nil %>
```

Comment:
I know this isn't part of the PR, but I'm not sure what this line is for. Do we need it?

Reply:
The markdown editor tries to emulate the textarea component in terms of how we structure the attributes when rendering a component (as shown in the example below).
```
    <%= render "components/markdown-editor", {
      textarea: {
        label: {
          text: schema.label
        }
      }
    } %>
```
The markdown editor is different from a textarea component by having the preview toolbar between the label element and the textarea element. Because of this, we don't render the label as part of the textarea component, but separately (L8-L11) then we remove the label from the textarea object (that stands as parameter for the textarea component).

Hope that makes sense.

Reply:
I think I see the aim, but I still don't understand if we actually need that line or not. Does the code work without it?

Reply:
We need it otherwise we'll have two labels. We need to stop the label going to the textarea component and render the 2nd time.

Reply:
Dropped the approach when we reuse the label parameter passed to the textarea component and moved into defining a separate label for the markdown editor component.